### ISSUE TITLE
Integer Type in Unique Composite Key Does Not Work

### ISSUE THREAD

**Issue by ashih42** [prisma/prisma#4296]
Hello,

I am using the latest `@prisma/client` and `@prisma/cli` version 2.11.0.

The following Schema defines `Cat` with a foreign key to `Workspace`, and it has a unique composite key combining the 2 fields `api_id` and `workspace_id`.

```
model Cat {
  id  String  @default(cuid()) @id

  api_id  Int

  workspace  Workspace  @relation(fields: [workspace_id], references: [id])
  workspace_id  String

  @@unique([api_id, workspace_id])
}

model Workspace {
  id  String  @default(cuid()) @id

  name  String

  cats  Cat[]
}
```

**Bug Example 1:**
After `prisma generate`, the following code would refuse to run with the compile-time error: `Type 'number' is not assignable to type 'string'`

```
const my_cat = await PRISMA_CLIENT.cat.findOne({
  where: {
    api_id_workspace_id: {
      api_id: 12345,
      workspace_id: 'a_valid_workspace_id',
    },
  },
});
```

The expected string type can be traced to these lines in the generated `index.d.ts`:
```
export type Api_idWorkspace_idCompoundUniqueInput = {
  api_id: string
  workspace_id: string
}
```

**Bug Example 2:**
After converting the `api_id` integer to string value, the following code would run
```
const my_cat = await PRISMA_CLIENT.cat.findOne({
  where: {
    api_id_workspace_id: {
      api_id: '12345',
      workspace_id: 'a_valid_workspace_id',
    },
  },
});
```

and throw this runt-ime error:
```
ERROR: PrismaClientUnknownRequestError2:
Invalid `prisma.cat.findOne()` invocation:


  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("22P03"), message: "incorrect binary data format in bind parameter 1", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("postgres.c"), line: Some(1740), routine: Some("exec_bind_message") }) }) })
    at PrismaClientFetcher.request (/home/ash/profits-backend/node_modules/@prisma/client/runtime/index.js:78786:15)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  clientVersion: '2.11.0'
}
```

**Bug Example 3:**
If you provide a ***non-integer*** string value for `api_id`, the code runs just fine (in this example it returns `null`).
```
const my_cat = await PRISMA_CLIENT.cat.findOne({
  where: {
    api_id_workspace_id: {
      api_id: 'what',
      workspace_id: 'a_valid_workspace_id',
    },
  },
});
```

I just want to use integer values as part of my unique composite key.  Please advise.  Thank you for your time and consideration.

**Comment by divyenduz** [comment]
@ashih42 Hey, I am unable to reproduce this. I tried the bug example 1 but it did not throw and just returned `{ my_cat: null }`

Can you please provide a reproduction repo?

**Comment by ashih42** [comment]
@divyenduz Thank you for checking in.  I realized my previous post was insufficient to produce the bug.  I have realized the bug was due to name collision of the intermediate variable `api_id_workspace_id`.  I have produced the bug in this repo:

**Comment by pantharshit00** [comment]
I can confirm this name collision with the reproduction that you provided @ashih42 

Thanks for the detailed reproduction repository.

**Comment by dpetrick** [comment]
A fix will be available in 2.14.
